kong/plugins/datadog/log.lua

```diff
@@ -0,0 +1,71 @@
+local statsd_logger = require "kong.plugins.datadog.statsd_logger"
+
+local ngx_log = ngx.log
+local ngx_timer_at = ngx.timer.at
+local string_gsub = string.gsub
+
+local _M = {}
+
+local METRICS = { "request_count", "latency", "request_size", "status_count" }
+
+local function request_counter(api_name, logger)
+  local stat = api_name..".request.count"
+  logger:counter(stat, 1, 1)
+end
+
+local function status_counter(api_name, message, logger)
+  local stat = api_name..".request.status."..message.response.status
+  logger:counter(stat, 1, 1)
+end
+
+local function request_size_gauge(api_name, message, logger)
+  local stat = api_name..".request.size"
+  logger:gauge(stat, message.request.size, 1)
+end
+
+local function latency_gauge(api_name, message, logger)
+  local latency = message.response.response_received - message.started_at
```

Comment:
Here I am looking for total time taken by proxy + upstream server to process the request and send response back to client, basically latency.